crates/holochain/tests/agent_scaling.rs

```diff
@@ -0,0 +1,176 @@
+use hdk3::prelude::Links;
+use holochain::test_utils::host_fn_api;
+use holochain::{
+    core::ribosome::ZomeCallInvocation, test_utils::conductor_setup::ConductorTestData,
+};
+use holochain_keystore::keystore_actor::KeystoreSenderExt;
+use holochain_serialized_bytes::prelude::*;
+use holochain_state::test_utils::test_environments;
+use holochain_types::dna::{DnaDef, DnaFile};
+use holochain_wasm_test_utils::TestWasm;
+use holochain_zome_types::{ExternInput, ZomeCallResponse};
+use unwrap_to::unwrap_to;
+
+/// A single link with an AgentPubKey for the base and target is committed by
+/// one agent, and after a delay, all agents can get the link
+#[tokio::test(threaded_scheduler)]
+async fn many_agents_can_reach_consistency_agent_links() {
+    const NUM_AGENTS: usize = 2;
+    let consistency_delay = std::time::Duration::from_secs(50);
+
+    let envs = test_environments();
+    let zomes = vec![TestWasm::Link];
+
+    let dna_file = DnaFile::new(
+        DnaDef {
+            name: "conductor_test".to_string(),
+            uuid: "ba1d046d-ce29-4778-914b-47e6010d2faf".to_string(),
+            properties: SerializedBytes::try_from(()).unwrap(),
+            zomes: zomes.clone().into_iter().map(Into::into).collect(),
+        },
+        zomes.into_iter().map(Into::into),
+    )
+    .await
+    .unwrap();
+
+    let mut agents = Vec::with_capacity(NUM_AGENTS);
+
+    for _ in 0..NUM_AGENTS {
+        agents.push(
+            envs.keystore()
+                .generate_sign_keypair_from_pure_entropy()
+                .await
+                .unwrap(),
+        )
+    }
+
+    let (conductor, cell_ids) =
+        ConductorTestData::new(envs, vec![dna_file], agents, Default::default()).await;
+
+    let cell_ids = cell_ids.values().next().unwrap();
+    let committer = conductor.call_data(&cell_ids[1]).unwrap();
+    let base = cell_ids[0].agent_pubkey().clone();
+    let target = cell_ids[1].agent_pubkey().clone();
+
+    // let base = host_fn_api::commit_entry(
+    //     &committer.env,
+    //     committer.call_data(TestWasm::Link),
+    //     Entry::app(().try_into().unwrap()).unwrap(),
+    //     "path",
+    // )
+    // .await;
+    // let target = base;
+
+    host_fn_api::create_link(
+        &committer.env,
+        committer.call_data(TestWasm::Link),
+        base.clone().into(),
+        target.into(),
+        ().into(),
+    )
+    .await;
```

Comment:
There's nothing here to publish the links. `host_fn_api` just calls the host function but when create_link is called from a zome call the publish handle is called. You can get hold of the trigger handle from the test data. Or just write a test zome to call create link.

Reply:
Wonder if we should add that to the machinery that calls into the conductor, to make it run the actual workflow and do the trigger instead of just doing a call. Do we ever want a test that just makes a call without doing the whole workflow?